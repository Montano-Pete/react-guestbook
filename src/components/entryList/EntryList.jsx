import useEntry from '../../hooks/useEntry';
import Entry from '../entry/Entry';

function EntryList() {
  const { entries } = useEntry();

  return (
    <>
      <ul>
        {entries.map((entry) => (
          <li key={`${entry.name}`}>
            <Entry entry={entry} />
          </li>
        ))}
      </ul>
    </>
  );
}

export default EntryList;
