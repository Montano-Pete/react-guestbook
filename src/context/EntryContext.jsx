import { createContext, useState } from 'react';

const EntryContext = createContext();

const EntryProvider = ({ children }) => {
  const [entry, setEntry] = useState();

  return (
    <EntryContext.Provider value={{ entry, setEntry }}>
      {children}
    </EntryContext.Provider>
  );
};

export { EntryProvider, EntryContext };
