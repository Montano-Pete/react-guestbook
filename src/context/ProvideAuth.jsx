import React from 'react';
import { createContext, useState } from 'react';

const authContext = createContext();

function ProvideAuth({ children }) {
  const [user, setUser] = useState(null);

  const login = (email, password) => {
    const loginSuccessful =
      email === process.env.AUTH_EMAIL &&
      password === process.env.AUTH_PASSWORD;
    if (loginSuccessful) setUser({ email });
    return loginSuccessful;
  };

  const logout = (callback) => {
    setUser(null);
    callback();
  };

  return (
    <authContext.Provider value={{ user, login, logout }}>
      {children}
    </authContext.Provider>
  );
}

export { authContext, ProvideAuth };
