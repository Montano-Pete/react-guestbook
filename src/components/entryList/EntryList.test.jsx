import React from 'react';
import { render } from '@testing-library/react';
import { EntryProvider } from '../../context/EntryContext';
import EntryList from './EntryList';

it('should render an entry list', () => {
  const { container } = render(
    <EntryProvider>
      <EntryList />
    </EntryProvider>
  );
  expect(container).toMatchSnapshot();
});
