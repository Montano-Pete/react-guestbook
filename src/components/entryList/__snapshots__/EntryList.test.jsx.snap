// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an entry list 1`] = `
<div>
  <ul />
</div>
`;
